### PR TITLE
HOCS-4221: Add ability to transfer BF cases to SMC

### PR DESCRIPTION
**Pull request by JosephRogerson**

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.api.CaseDataService;
+import uk.gov.digital.ho.hocs.casework.api.CorrespondentService;
+import uk.gov.digital.ho.hocs.casework.api.factory.CaseCopy;
+import uk.gov.digital.ho.hocs.casework.domain.model.CaseData;
+
+@Service
+@CaseCopy(fromCaseType = "BF", toCaseType = "SMC")
+public class CopyBFToSMC extends AbstractCaseCopyStrategy implements CaseCopyStrategy {
```
Comment (gary-newbolt) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java:12]: Are you writing any tests for this strategy @JosephRogerson?

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            "OwningCSU",
+            "ComplainantDOB",
+            "ComplainantHORef",
+            "ComplainantPortRef",
+            "ComplainantCompanyName",
+            "ComplainantNationality",
+            "ComplainantGender",
+            "Region",
+            "BusArea",
+            "EnquiryReason1",
```
Comment (gary-newbolt) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java:28]: You'll need the most up to date info from BF... the `EnquiryReasonX` fields no longer exist in BF/BF2 and are now called `ComplaintReason...` and have 5 of them, as well as associated `Other` fields. 

That being said, SMC has no concept of these fields so do they need to be copied? You might want to wait for the meeting on the transfer items

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public void copyCase(CaseData fromCase, CaseData toCase) {
+
+        // copy clob details
+        copyClobData(fromCase, toCase, DATA_CLOB_KEYS);
+        toCase.update("PreviousCaseReference", fromCase.getReference());
+        caseDataService.updateCaseData(toCase.getUuid(), null, toCase.getDataMap());
+
+        // Correspondents include the primary_correspondent_uuid
+        correspondentService.copyCorrespondents(fromCase.getUuid(), toCase.getUuid());
+
```
Comment (gary-newbolt) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java:55]: Do we need to copy documents?

**Review by gary-newbolt — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java (hunk 1 above)
Comment (JosephRogerson) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java:12]: I've added a test for this strategy

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java (hunk 2 above)
Comment (JosephRogerson) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java]: I shouldn't have added these fields as they come from triage and it's only possible to transfer before triage. I've taken them and any other non-registration fields out

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import uk.gov.digital.ho.hocs.casework.domain.model.CaseData;
+
+@Service
+@CaseCopy(fromCaseType = "BF", toCaseType = "SMC")
+public class CopyBFToSMC extends AbstractCaseCopyStrategy implements CaseCopyStrategy {
+
+    private static final String[] DATA_CLOB_KEYS = {
+            "Channel",
+            "CompType",
+            "CaseSummary",
```
Comment (gary-newbolt) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java:17]: After you merge in the latest changes from main, can you review these? The mapping I think should be to different vars... Work done on HOCS-4590 separated the var names that these map too.

Comment (JosephRogerson) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/factory/strategies/CopyBFToSMC.java]: I believe the only one of these that also exists in SMC cases is the summary, so I've added a mapping for this

**Review by gary-newbolt — APPROVED** [review]

**Pull request merged**